### ISSUE TITLE
GraphIdentifiers don't seem to be cleaned-up

### ISSUE THREAD

**Issue by jbehrens94** [Swinject/Swinject#510]
Hi all,

I'm working on an app extension with extremely limited resources in terms of memory. I noticed that Swinject is claiming a lot of memory through `GraphIdentifier`, it'll increase quite a lot when doing regular actions throughout the extension.

Am I using Swinject wrong, or what should I do to minimise that amount in memory?

**Comment by jbehrens94** [comment]
@mpdifran Who's able to help me on this? I don't know who the current contributors are, because I saw in an issue from 2020 that it looked like Swinject isn't maintained actively.

**Comment by jbehrens94** [comment]
CC @yoichitgy Maybe you can help?

**Comment by jbehrens94** [comment]
@mpdifran @yoichitgy Can you confirm that my previously sent message contains a correct setup for Swinject?

**Comment by jbehrens94** [comment]
![Schermafbeelding 2022-07-27 om 11 07 52](https://user-images.githubusercontent.com/5713854/181208876-4db98916-32fa-4299-8133-b314ecafba79.png)
![Schermafbeelding 2022-07-27 om 11 07 03](https://user-images.githubusercontent.com/5713854/181208863-8399c39c-7501-4cae-a583-cf53d848942f.png)

The first screenshot is a clean start, the second screenshot is after an action in the app extension is finished. The memory Swinject is claiming seems to only be growing.

**Comment by mpdifran** [comment]
@jbehrens94 yeah your setup looks correct, I don't see anything wrong with it.

**Comment by jbehrens94** [comment]
Same question for the Swinject.(Weak..) things

**Comment by bbrk24** [comment]
I investigated this a little, and from what I can tell, the `GraphIdentifier` and `Weak<Any>` objects are being held onto by `GraphStorage`. Perhaps instead of `[GraphIdentifier: Weak<Any>]`, it needs some kind of `WeakDictionary<GraphIdentifier, Any>`.
